frontend_csv/src/parser/parser.rs

```diff
@@ -0,0 +1,65 @@
+use super::ast_msg;
+use super::ast_msg::AstMsgLineType;
+// use super::ast_msg::AstMsgTrait;
+// use super::ast_msg::AstMsgType;
+// use super::ast_msg::InitMsg;
+use crate::scanner::token::CSVToken;
+use crate::scanner::token::CSVTokenType;
+/// Core implementation of the parser
+
+pub struct Parser {}
+
+impl Parser {
+    pub fn new() -> Parser {
+        return Parser {};
+    }
+
+    pub fn parse_token(&mut self, source: &Vec<CSVToken>) -> Vec<(AstMsgLineType, Vec<String>)> {
```

Comment:
Why do you return a vector of `Vec<(AstMsgLineType, Vec<String>)>`? What is the String vector?

Reply:
the string vector is the value, for example ["msgtype", "Init", 16] is the value, while the line type is AstMsgLineType:msgtype... need those value to make check if the interger value is 16 or store the "init" for future lines reference checking